api/src/routes/protected/curriculum-section.ts

```diff
@@ -0,0 +1,136 @@
+import { type FastifyPluginCallbackTypebox } from '@fastify/type-provider-typebox';
+import { CompletedChapter, CompletedModule } from '@prisma/client';
+
+import * as schemas from '../../schemas';
+
+const getInformationForUpdate = ({
+  id: sectionId,
+  completedItems,
+  completedDate
+}: {
+  id: undefined | string;
+  completedItems: CompletedModule[] | CompletedChapter[];
+  completedDate: number;
+}) => {
+  const alreadyCompletedSection = sectionId
+    ? completedItems.find(({ id }) => sectionId === id)
+    : undefined;
+
+  const newCompletedSection =
+    !sectionId || alreadyCompletedSection
+      ? completedItems
+      : {
+          push: {
+            id: sectionId,
+            completedDate
+          }
+        };
```

Comment:
```suggestion
  if (!sectionId) {
    return {
      alreadyCompletedSection: undefined,
      newCompletedSection: completedItems
    };
  }

  const alreadyCompletedSection = completedItems.find(
    ({ id }) => sectionId === id
  );

  const newCompletedSection = alreadyCompletedSection
    ? completedItems
    : {
        push: {
          id: sectionId,
          completedDate
        }
      };
```
Preference, I guess, but this means I can forget about `sectionId` a bit earlier so `newCompletedSection` is a little easier to understand.